ui/src/dashboards/components/TemplateVariableManager.js

```diff
@@ -0,0 +1,235 @@
+import React, {Component, PropTypes} from 'react'
+import classNames from 'classnames'
+import uuid from 'node-uuid'
+
+import TemplateVariableTable
+  from 'src/dashboards/components/TemplateVariableTable'
+
+import {TEMPLATE_VARIABLE_TYPES} from 'src/dashboards/constants'
+
+const TemplateVariableManager = ({
+  onClose,
+  onEditTemplateVariables,
+  source,
+  templates,
+  onRunQuerySuccess,
+  onRunQueryFailure,
+  onSaveTemplatesSuccess,
+  onAddVariable,
+  onDelete,
+  tempVarAlreadyExists,
+  isEdited,
+}) => (
+  <div className="template-variable-manager">
+    <div className="template-variable-manager--header">
+      <div className="page-header__left">
+        Template Variables
+      </div>
+      <div className="page-header__right">
+        <button className="btn btn-primary btn-sm" onClick={onAddVariable}>
+          Add Variable
+        </button>
+        <button
+          className={classNames('btn btn-success btn-sm', {
+            disabled: !isEdited,
+          })}
+          onClick={onEditTemplateVariables(templates, onSaveTemplatesSuccess)}
+        >
+          Save Template
+        </button>
+        <span
+          className="icon remove"
+          onClick={() => onClose(isEdited)}
+          style={{cursor: 'pointer'}}
+        />
+      </div>
+    </div>
+    <div className="template-variable-manager--body">
+      <TemplateVariableTable
+        source={source}
+        templates={templates}
+        onRunQuerySuccess={onRunQuerySuccess}
+        onRunQueryFailure={onRunQueryFailure}
+        onDelete={onDelete}
+        tempVarAlreadyExists={tempVarAlreadyExists}
+      />
+    </div>
+  </div>
+)
+
+class TemplateVariableManagerWrapper extends Component {
```

Comment:
Is it correct to interpret this Wrapper as a shim with Redux-like behavior? So that lower-level components don't need to do state management